### PR TITLE
More updates

### PR DESCRIPTION
**Pull request by katzdm**
Staring an R6 with some more updates. Finally passed 100pgs, it's about time.

### CODE REVIEW

File: 2996_reflection/reflection.md [hunk 1]
```diff
@@ -2358,9 +2385,14 @@ namespace std::meta {
   consteval auto data_member_spec(info type_class,
                                   data_member_options_t options = {}) -> info;
   template <reflection_range R = initializer_list<info>>
-  consteval auto define_class(info type_class, R&&) -> info;
+    consteval auto define_class(info type_class, R&&) -> info;
+
+  // @[define_static_array](#define_static_array)@
+  template <typename T, input_range<T> R = initializer_list<T>>
+    consteval auto define_static_array(R&& args) -> span<>;
```
Comment (brevzin) [inline 2996_reflection/reflection.md:2392]: This declaration is broken, what's the intent here?

Also, there's no `#define_static_array` section, no corresponding description of this function, and no wording. Split it up into its own PR?

File: 2996_reflection/reflection.md [hunk 2]
```diff
@@ -3122,6 +3156,31 @@ has its linkage determined as follows:
...
+Add a bullet to paragraph 13:
+
+::: std
+
+[13]{.pnum} A declaration `$D$` _names_ an entity `$E$` if
+
+* `$D$` contains a _lambda-expression_ whose closure type is `$E$`,
+* [`$D$` contains a _splice-specifier_ designating `$E$`,]{.addu}
+
+...
```
Comment (brevzin) [inline 2996_reflection/reflection.md:3168]: When adding bullets, always number them. And them might as well provide the other two bullets for context.

```suggestion
* [13.1]{.pnum} `$D$` contains a _lambda-expression_ whose closure type is `$E$`,
* [13.1+]{.pnum} [`$D$` contains a _splice-specifier_ designating `$E$`,]{.addu}
* [13.2]{.pnum} `$E$` is not a function or function template and `$D$` contains an *id-expression*, *type-specifier*, *nested-name-specifier*, *template-name*, or *concept-name denoting* `$E$`, or
* [13.#]{.pnum} `$E$` is a function or function template and `$D$` contains an expression that names `$E$` ([basic.def.odr]) or an *id-expression* that refers to a set of overloads that contains `$E$`.
```

File: 2996_reflection/reflection.md [hunk 3]
```diff
@@ -3122,6 +3156,31 @@ has its linkage determined as follows:
...
+
+Extend the definition of _TU-local_ values and objects to include reflections:
+
+::: std
+
+[16]{.pnum} A value or object is _TU-local_ if either
+
+* [16.1]{.pnum} it is, or is a pointer to, a TU-local function or the object associated with a TU-local variable, [or]{.rm}
+* [[16.2]{.pnum} it is a reflection representing a TU-local entity, or]{.addu}
+* [16.3]{.pnum} it is an object of class or array type and any of its subobjects or any of the objects or functions to which its non-static data members of reference type refer is TU-local and is usable in constant expressions.
```
Comment (brevzin) [inline 2996_reflection/reflection.md:3180]: Keep the existing numbering (so the middle one becomes `16.1*` or something

File: 2996_reflection/reflection.md [hunk 4]
```diff
@@ -26,6 +26,11 @@ tag: constexpr
 
 # Revision History
 
+Since [@P2996R5]:
+
+* removed `op_`-prefixes from `std::meta::operators` enumerators.
```
Comment (brevzin) [inline 2996_reflection/reflection.md:31]: I'd prefer to back this out.

File: 2996_reflection/reflection.md [hunk 5]
```diff
@@ -2074,6 +2081,26 @@ This paper is proposing that:
...
+
+In light of the above, our position is that `type_of` should never return aliases: That is, `StrTy` represents `std::basic_string<char, std::allocator<char>>`. We believe that it would be desireable to in the future introduce an `aliased_type_of` function capable of returning representations of both `std::string` and `const std::string &` for `StrTy` and `RefTy` respectively - but this requires both discussions with implementers, and likely new wording technology for the Standard. To avoid jeopardizing the goal declared by the title of this paper, we are not proposing such a function at this time.
```
[the remark this answers is not in the document]
Comment (katzdm) [reply 2996_reflection/reflection.md:2102]: > "desirable" (no extra e).

Gets me every time.

> Also, are we proposing aliased_type_of or not?

No, we are not at this time (for the reasons stated above).

> Like is there no way to get std::string from a reflection of the variable Str?

Correct - though it's possible to compare `type_of(^V) == ^std::string` to know that it's the same underlying type.

> But does anybody actually want those aliases?

Good question.
- `substitute`, I think, _must_ return `some_trait_t<type>`. Consider:

```cpp
template <typename T> class S { };
template <typename T> using A = S<T>;

constexpr auto r = substitute(^A, {^int});
```

Part of the contract of `substitute` is that it only substitutes the template, but does not instantiate it. I think that forming the template-id `S<T>` would go beyond substitution of `A`.

- For `nested_name` (which I don't think we should tackle for P2996), I really have no idea. I don't know whether I find that to be more similar to `type_of` or to `^some_trait<type>::type`.

File: 2996_reflection/reflection.md (hunk 1 above)
Comment (katzdm) [reply 2996_reflection/reflection.md:2392]: Oh man, that's super broken lol. Sure, we can do a separate PR.

File: 2996_reflection/reflection.md [hunk 6]
```diff
@@ -2358,9 +2385,14 @@ namespace std::meta {
   consteval auto data_member_spec(info type_class,
                                   data_member_options_t options = {}) -> info;
   template <reflection_range R = initializer_list<info>>
-  consteval auto define_class(info type_class, R&&) -> info;
+    consteval auto define_class(info type_class, R&&) -> info;
+
+  // @[define_static_array](#define_static_array)@
+  template <typename T, input_range<T> R = initializer_list<T>>
+    consteval auto define_static_array(R&& args) -> span<>;
+  template <input_range R>
```
[the remark this answers is not in the document]
Comment (katzdm) [reply 2996_reflection/reflection.md:2393]: Delorted from this PR instead.

File: 2996_reflection/reflection.md (hunk 2 above)
Comment (katzdm) [reply 2996_reflection/reflection.md:3168]: Done, thanks.

File: 2996_reflection/reflection.md (hunk 3 above)
Comment (katzdm) [reply 2996_reflection/reflection.md:3180]: Done, used `16.1+` like you did above.

File: 2996_reflection/reflection.md [hunk 7]
```diff
@@ -3863,6 +3922,15 @@ Extend *template-argument-equivalent* to handle `std::meta::info`:
 * [2.5]{.pnum} [...]
 :::
 
+### [temp.alias]{.sref} Alias templates {-}
+
+Extend paragraph 2 to enable reflection of alias template specializations.
+
+::: std
+[2]{.pnum} [When]{.rm} [Except when used as the operand of a `$reflect-expression$`,]{.addu} a `$template-id$` refer[ring]{.addu}[s]{.rm} to a specialization of an alias template[, it]{.rm} is equivalent to the associated type obtained by substitution of its `$template-arguments$` for the `$template-parameter$`s in the `$defining-type-id$` of the alias template.
```
[the remark this answers is not in the document]
Comment (katzdm) [reply 2996_reflection/reflection.md:3930]: Always ruining my fun.

File: 2996_reflection/reflection.md (hunk 4 above)
Comment (katzdm) [reply 2996_reflection/reflection.md]: Done.

**Review by brevzin — APPROVED** [review]
